### PR TITLE
First pass at methods

### PR DESCRIPTION
**Pull request by briansteffens**
Sorry @eatonphil for the ```this``` keyword :smile:

### CODE REVIEW

**Comment by eatonphil** [comment]
No committed test / example?

**Comment by briansteffens** [comment]
methods.bs has an example, no tests yet though

**Comment by eatonphil** [comment]
Oh I missed that. It was hard to tell it wasn't D at this point

File: methods.bs [hunk 1]
```diff
@@ -0,0 +1,33 @@
+struct rect
+{
+    u64 width;
+    u64 height;
+}
+
+u64 rect::equals(rect* other)
```
Comment (eatonphil) [inline methods.bs:7]: You have `bool`s and you still return `u64` here? -1 :p

Comment (briansteffens) [reply methods.bs:7]: Yeah something's screwy with the bools and/or the ```&&``` operator lol

Comment (eatonphil) [reply methods.bs:7]: heh

**Pull request merged**